src/ListenerCollectionFactory.php

```diff
@@ -22,54 +26,31 @@ public function __construct(Injector $injector, ContainerInterface $container)
 
     /**
      * @param array $eventListeners Event listener list in format ['eventName1' => [$listener1, $listener2, ...]]
-     * @param bool $precheck If true, all the listeners will be checked strictly if they are valid callables
-     *                       before attaching to the ListenerCollection
      *
      * @return ListenerCollection
      */
-    public function create(array $eventListeners, bool $precheck = false): ListenerCollection
+    public function create(array $eventListeners): ListenerCollection
     {
         $listenerCollection = new ListenerCollection();
 
         foreach ($eventListeners as $eventName => $listeners) {
-            if (!is_string($eventName)) {
+            if (!is_string($eventName) || !class_exists($eventName)) {
```

Comment:
Such check exists in `ListenerConfigurationChecker`. Should we duplicate it here?